app/Repositories/GitPathsRepository.php

```diff
@@ -31,16 +30,15 @@ public function __construct($path)
      */
     public function dirty()
     {
-        $process = tap(new Process(['git', 'status', '--short', '--', '*.php']))->run();
+        // Get ready-to-use paths of files that were added, modified, moved or renamed
+        $process = tap(new Process(['git', 'diff', 'HEAD', '--name-only', '--diff-filter=ACMR', '--', '*.php']))->run();
 
         if (! $process->isSuccessful()) {
             abort(1, 'The [--dirty] option is only available when using Git.');
         }
 
-        $dirtyFiles = collect(preg_split('/\R+/', $process->getOutput(), flags: PREG_SPLIT_NO_EMPTY))
-            ->mapWithKeys(fn ($file) => [substr($file, 3) => trim(substr($file, 0, 3))])
-            ->reject(fn ($status) => $status === 'D')
-            ->map(fn ($status, $file) => $status === 'R' ? Str::after($file, ' -> ') : $file)
+        $dirtyFiles = collect(explode("\n", $process->getOutput()))
+            ->filter()
             ->map(fn ($file) => $this->path.DIRECTORY_SEPARATOR.$file)
```

Comment:
I left it as it was, but I'm wondering if it's really needed. Relative paths should work as arguments for analysis. But when I remove it, no dirty files are found because of intersection check (below). Is it really required to compare diff result with list of files in the directory? Since it's result of Git, we can be sure files found are from this repository where analysis is going to be run, right? 🤔

Reply:
@Wirone I could think of cases where there is files committed yet that I don't want to fix via a CS fixer

Reply:
@theofidry yes, totally understandable, but with this particular implementation it only intersects paths from `git diff` with all files' paths within `$this->path` (project dir?), which does not add real value IMHO. It would make more sense if intersection would be done with paths configured for analysis, but it's too complex I think.